### DISCUSSION TITLE
janitor is not cleaning up the expired access_tokens generated using client_credentials flow

### DISCUSSION THREAD

**Discussion by sagarshah1983** [ory/hydra#3781] [Q&A] [answered]
We are using hydra 2.2.0 and we are noticing growth in table hydra_oauth2_access.
After looking up further, we found that even when we run janitor command, its not cleaning up the tokens from hydra_oauth2_access table even though those are expired (we are using default hydra level expiry as 30m and there's no client specific expiry being set up), which were generated and issued to a client using client_credentials workflow.
Is that an expected behavior?

Please advise.

Regards,
Sagar

**Comment by sagarshah1983** [comment]
Related bug has been created for the same - #3784

**Comment by vinckr** [comment, marked as answer]
Hello @sagarshah1983,
have you had a look at the "Ory Enterprise License" (OEL): https://www.ory.sh/docs/self-hosted/oel/quickstart ? 
This version of Ory Hydra has "Automatic clean up of stale records" - no Hydra Janitor needed.
